### PR TITLE
Feature/138523 chargelinks test data

### PR DESCRIPTION
**Pull request by kristianschneider**
This implements charge links and charge prices and introduced a more generics approach for handling parsing

### CODE REVIEW

**Review by johevemi — APPROVED** [review]

**Review by johevemi — APPROVED** [review]

File: source/test-data-generation/GreenEnergyHub.Aggregation.TestData.Application/Parsers/ChargeLinkTestDataParser.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+namespace GreenEnergyHub.Aggregation.TestData.Application.Parsers
+{
+    public class ChargeLinkTestDataParser : TestDataParserBase<ChargeLink>, ITestDataParser
+    {
+        public ChargeLinkTestDataParser(IMasterDataStorage masterDataStorage, GeneratorSettings generatorSettings)
+            : base(masterDataStorage, generatorSettings)
+        {
+        }
+
+        public override string FileNameICanHandle => "ChargeLinks.csv";
```
Comment (ZavezX) [inline source/test-data-generation/GreenEnergyHub.Aggregation.TestData.Application/Parsers/ChargeLinkTestDataParser.cs:29]: Maybe we should move the file name to an app setting as these filenames could potentially change or be different in the first place.

File: source/test-data-generation/GreenEnergyHub.Aggregation.TestData.Application/Parsers/ChargePriceTestDataParser.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+namespace GreenEnergyHub.Aggregation.TestData.Application.Parsers
+{
+    public class ChargePriceTestDataParser : TestDataParserBase<ChargePrices>, ITestDataParser
+    {
+        public ChargePriceTestDataParser(IMasterDataStorage masterDataStorage, GeneratorSettings generatorSettings)
+            : base(masterDataStorage, generatorSettings)
+        {
+        }
+
+        public override string FileNameICanHandle => "ChargePrices.csv";
```
Comment (ZavezX) [inline source/test-data-generation/GreenEnergyHub.Aggregation.TestData.Application/Parsers/ChargePriceTestDataParser.cs:28]: Same goes here.

File: source/test-data-generation/GreenEnergyHub.Aggregation.TestData.Application/Parsers/ChargeTestDataParser.cs [hunk 3]
```diff
@@ -12,37 +12,20 @@
...
+    public class ChargeTestDataParser : TestDataParserBase<Charge>, ITestDataParser
     {
-        public ChargeTestDataParser(IMasterDataStorage masterDataStorage)
-            : base(masterDataStorage)
+        public ChargeTestDataParser(IMasterDataStorage masterDataStorage, GeneratorSettings generatorSettings)
+            : base(masterDataStorage, generatorSettings)
         {
         }
 
         public override string FileNameICanHandle => "charges.csv";
```
Comment (ZavezX) [inline source/test-data-generation/GreenEnergyHub.Aggregation.TestData.Application/Parsers/ChargeTestDataParser.cs:29]: Here as well.

**Review by mknic — APPROVED** [review]

**Review by ZavezX — CHANGES_REQUESTED** [review]

**Comment by ZavezX** [comment]
@kristianschneider I was a bit late to the review party, but I have some comments on this one.

**Pull request merged**